### PR TITLE
fix(2690): Get file as buffer

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Bitbucket SCM is unable to fetch `screwdriver.yaml` after version >=4.5.0. This is likely related to the change to using screwdriver-request package.

## Objective

This PR switches to taking in file as buffer to avoid screwdriver-request package [throwing error on 200](https://github.com/sindresorhus/got/blob/0703318d3bb6cc1340614612d2ae14b23f651242/source/core/response.ts#L141). Also adds try/catch around decorateAuthor and getFile.
Also adds try/catch to request calls.

_Note: decorateAuthor will return empty for now because of bitbucket's switch from username -> UUID._
Announcement: https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-changes-gdpr/
Related issues: https://stackoverflow.com/questions/53983942/how-to-find-bitbucket-account-uuid, https://community.atlassian.com/t5/Bitbucket-questions/Find-users-in-bitbucket-based-on-nickname/qaq-p/1041707

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/2690

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -426,38 +438,39 @@ class BitbucketScm extends Scm {
...
-            // have to return a simple generated decoration result to allow all builds to function.
-            // We will only allow this if the username is not a {uuid} pattern. Since if this is a {uuid}
-            // pattern, this likely is a valid 404.
             return {
-                url: '',
-                name: username,
-                username,
-                avatar: ''
+                url: body.links.html.href,
+                name: body.display_name,
```
Comment (pritamstyz4ever) [inline index.js:454]: Do we need any checks for undefined here?

File: index.js [hunk 2]
```diff
@@ -712,13 +739,16 @@ class BitbucketScm extends Scm {
...
+            return this.breaker.runCommand(options);
+        } catch (err) {
+            if (err.statusCode !== 201 && err.statusCode !== 200) {
+                logger.error('Failed to getFile: ', err);
+                throw err;
             }
 
-            return response;
-        });
+            return undefined;
```
Comment (pritamstyz4ever) [inline index.js:750]: can we return the error for other status codes too?

**Review by pritamstyz4ever — COMMENTED** [review]
Minor comments

File: index.js (hunk 2 above)
Comment (tkyi) [reply index.js:750]: changed to match: https://github.com/screwdriver-cd/scm-github/blob/master/index.js#L1095

File: index.js (hunk 1 above)
Comment (tkyi) [reply index.js]: updated to use hoek

**Review by jithine — APPROVED** [review]

**Review by pritamstyz4ever — APPROVED** [review]

**Pull request merged**